packages/api/cms-api/src/dam/files/files.service.ts

```diff
@@ -240,6 +240,95 @@ export class FilesService {
+                        ...image,
+                        width: image.height,
+                        height: image.width,
+                    };
+                }
+            } catch (e) {
+                // empty
+            }
+
+            if (
+                image !== undefined &&
+                image.width &&
+                image.height &&
+                Math.round(((image.width * image.height) / 1000000) * 10) / 10 >= this.imgproxyConfig.maxSrcResolution
+            ) {
+                throw new CometImageResolutionException(`Maximal image resolution exceeded`);
+            }
+
+            if (folderId) {
+                const folder = await this.foldersService.findOneById(folderId);
+                if (!folder) throw new Error("Folder not found");
+                if (!this.contentScopeService.scopesAreEqual(folder.scope, scope)) {
+                    throw new Error("Folder scope doesn't match passed scope");
+                }
+            }
+
+            let exifData: Record<string, string | number | Uint8Array | number[] | Uint16Array> | undefined;
+            if (exifrSupportedMimetypes.includes(file.mimetype)) {
+                exifData = await exifr.parse(file.path);
+            }
+
+            const exisitingFile = await this.findOneByFilenameAndFolder({ filename: file.originalname, folderId });
+            if (!exisitingFile) throw new Error("File not found");
+
+            await this.blobStorageBackendService.upload(file, contentHash, this.config.filesDirectory);
+
+            // Check if the current file is the only one using the contentHash before deleting from blob storage
+            if (
+                (await withFilesSelect(this.filesRepository.createQueryBuilder("file"), { contentHash: exisitingFile.contentHash }).getResult())
+                    .length === 1
+            ) {
+                await this.blobStorageBackendService.removeFile(this.config.filesDirectory, createHashedPath(exisitingFile.contentHash));
+            }
+
+            if (
+                image &&
+                image.width &&
+                image.height &&
+                exisitingFile.image &&
+                (image.height !== exisitingFile.image.height || image.width !== exisitingFile.image.width)
+            ) {
+                exisitingFile.image.width = image.width;
+                exisitingFile.image.height = image.height;
+                exisitingFile.image.exif = exifData;
+
+                if (exisitingFile.image.cropArea.focalPoint !== FocalPoint.SMART) {
+                    exisitingFile.image.cropArea = { focalPoint: FocalPoint.SMART };
+                }
+            }
+
+            Object.assign(exisitingFile, {
+                size: file.size,
+                mimetype: file.mimetype,
+                contentHash,
+                ...assignData,
+            });
+
+            result = await this.save(exisitingFile);
+        } catch (e) {
+            rimraf.sync(file.path);
```

Comment:
Shouldn't this also be removed in the happy path?

Reply:
3769285903f5266a90ba48f8fd110f6a3679f9df